spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -104,95 +93,135 @@ function AuthenticateTx(msgs []Any, portId string) error {
+}
+
+
+// DeleteActiveChannelID removes the active channel keyed by the provided portID stored in state
+function (k Keeper) DeleteActiveChannelID(portID string) {
 }
+
 ```
 
-### Registering & Controlling flows
-**Active Channels**
+### Register & Controlling flows
+
+#### Register Account Flow
+
+To register an interchain account we require an off-chain process (relayer) to listen for `OnChannelOpenInit` events with the capability to finish a channel creation handshake on a given connection. 
+
+1. The controller chain binds a new IBC port with an id composed of the *source/counterparty connection-ids* & the *interchain account owner address*
+
+The IBC port identifier will look like this:
+```
+ics27-1.{connection-id}.{counterparty-connection-id}.{owner-account-address}
+```
+
+This port will be used to create channels between the controller & host chain for a specific owner/interchain account pair. Only the account with `{owner-account-address}` matching the bound port will be authorized to send IBC packets over channels created with `ics27-1.{connection-id}.{counterparty-connection-id}.{owner-account-address}`. It is up to each controller chain to enforce this port registration and access on the controller side. 
+
+2. The controller chain emits an event signaling to open a new channel on this port given a connection 
+3. A relayer listening for `OnChannelOpenInit` events will begin the channel creation handshake
+4. During the `OnChanOpenTry` callback on the host chain an interchain account will be registered and a mapping of the interchain account address to the owner account address will be stored in state (this is used for authenticating transactions on the host chain at execution time). 
+5. During the `OnChanOpenAck` callback on the controller chain a record of the interchain account address registered on the host chain during `OnChanOpenTry` is set in state with a mapping from portID -> interchain account address. See [version negotiation](#Version-Negotiation) section below for how to implement this
+6. During the `OnChanOpenAck` & `OnChanOpenConfirm` callbacks on the controller & host chains respectively, the [active-channel](#Active-Channels) for this interchain account/owner pair, is set in state
+
+#### Active Channels
 
 The controller and host chain should keep track of an `active-channel` for each registered interchain account. The `active-channel` is set during the channel creation handshake process. This is a safety mechanism that allows a controller chain to regain access to an interchain account on a host chain in case of a channel closing. 
 
-An active channel can look like this:
+An example of an active channel on the controller chain can look like this:
 
 
 ```typescript
 {
  // Controller Chain
- SourcePortId: `ics-27-0-0-cosmos1mjk79fjjgpplak5wq838w0yd982gzkyfrk07am`,
- SourceChannelId: `channel-1`,
+ SourcePortId: `ics27-<version>.<source-connection-id>.<counterparty-connection-id>.<owner-account-address>`,
```

Comment:
It may be confusing to mix the "source" terminology with "counterparty". Consider using _**one**_ of these pairs of terms:

- `sourcePortId` and `destinationPortId`
- `portId` and `counterpartyPortId`

This is also in line with other ICS-es ([example](https://github.com/cosmos/ibc/tree/b25d9b8b2bae739821703404266256bb9a741027/spec/core/ics-004-channel-and-packet-semantics)).